mathics/builtin/assignments/internals.py

```diff
@@ -608,12 +608,7 @@ def process_rhs_conditions(lhs, rhs, condition, evaluation):
 
 
 def process_tags_and_upset_dont_allow_custom(tags, upset, self, lhs, focus, evaluation):
-    # TODO: the following provides a hacky fix for 1259. I know @rocky loves
-    # this kind of things, but otherwise we need to work on rebuild the pattern
-    # matching mechanism...
```

Comment:
I note we didn't need to "work on rebuild the pattern matching mechanism". I point out partly because there is a tendency to exaggerate.

Instead, my take is that we needed to think about this more like it might be written if this code were written in WL itself and understand the WL OneIdentity better.

I am glad to see this hack get removed.

Reply:
It was not so bad as I thought then, but nevertheless I had to make changes on the mathics.core.pattern module. Maybe now we can rewrite this part better.